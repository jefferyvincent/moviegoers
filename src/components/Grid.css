.grid {
    display: flex;
    width: 100%;
    flex-wrap: wrap;
    justify-content: center;
    padding-top: 1rem;
    list-style: none;
}

.gid li {
    list-style: none;
}