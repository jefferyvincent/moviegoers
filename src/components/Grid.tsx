import React from 'react';

import './Grid.css';
import Movie from './Movie';

interface GridProps {
    movies: any;
}

const Grid: React.FC<GridProps> = ({movies}) => {
    const renderMovies = () => {
        if (movies.hasOwnProperty('results')) {
            return movies.results.map((item:any, key:number) => {
                return <li><Movie key={key} id={item.id} img={item.poster_path} title={item.title}/></li>  
            });
        } else {
            return <li><div className="loader">Loading...</div></li>
        }
    }
    return (
        <ul className="grid">
            {renderMovies() }
        </ul>
    )
}

export default Grid;