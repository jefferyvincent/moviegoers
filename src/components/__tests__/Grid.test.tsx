import React from 'react';
import { shallow } from 'enzyme';
import {render, cleanup, waitForElement } from 'react-testing-library';
import "jest-dom/extend-expect";
import axios from "axios";

import Grid from '../Grid';
afterEach(cleanup);

describe('<Grid/>', () => {
  let wrapper:any;
  let data = [{}];
     
  beforeEach(() => { wrapper = shallow(<Grid movies={data}/>); });

  it("renders", () => {
    const { asFragment } = render(<Grid movies={data} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders an grid', () => {
    expect(wrapper.find('ul.grid').length).toEqual(1);
  });
  
  it('renders an loader', () => {
    expect(wrapper.find('div.loader').length).toEqual(1);
  });

  /* 
    Need to mock Ajax somehow here???? Nothing is is working and __mocks__ / axios.js is throwing me errors. 
    Reference:
    https://www.leighhalliday.com/mocking-axios-in-jest-testing-async-functions
    https://www.leighhalliday.com/async-axios-react-testing-library
  */

  it('fetches and displays data', async () => {
    const {} = render(<Grid movies={data} />);
  });

});