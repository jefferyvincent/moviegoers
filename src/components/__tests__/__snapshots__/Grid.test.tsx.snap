// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid/> renders 1`] = `
<DocumentFragment>
  <ul
    class="grid"
  >
    <li>
      <div
        class="loader"
      >
        Loading...
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
